File: front-end/src/app/service-models/app/v1/client/leave-report-request-s-m.ts
import { BaseReportFilterSM } from './base-report-filter-s-m';
import { LeaveTypeSM } from '../../enums/leave-type-s-m.enum';

export class LeaveReportRequestSM extends BaseReportFilterSM {
    clientEmployeeUserId!: number;
    leaveCount!: number;
    leaveType!: LeaveTypeSM;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/LeaveReportRequestSM.cs
using Siffrum.Web.Payroll.ServiceModels.Enums;

namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class LeaveReportRequestSM : BaseReportFilterSM
    {
        public int ClientEmployeeUserId { get; set; }
        public int LeaveCount { get; set; }
        public LeaveTypeSM LeaveType { get; set; }

    }
}
